curator-framework/src/main/java/org/apache/curator/framework/api/AddWatchBuilder.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.curator.framework.api;
+
+import org.apache.zookeeper.AddWatchMode;
+
+public interface AddWatchBuilder extends AddWatchBuilder2
```

Comment:
Why do we need AddWatchBuilder2 and AddWatchBuilder separated? Why not merge them into one interface? Did you want to clean AddWatchBuilder from the parent interfaces?

Reply:
`AddWatchBuilder` has the "withMode" method which is optional.

Reply:
OK, it is just a little hard for me with using the same name but with 2 as suffix, but I don't have a better name to offer :)